src/main/java/fi/aalto/cs/apluscourses/intellij/model/IntelliJModelFactory.java

```diff
@@ -41,23 +39,30 @@ public Course createCourse(@NotNull String name,
       @NotNull List<Library> libraries,
```

Comment:
keeps growing :heart_eyes:, do you think the whole method or the at least the methods introduced in the anonymous classes can be tested (as this might become **very** expensive to debug)?

Reply:
This calls for big refactoring IMO. Such refactoring that makes it logic unit testable. Now it's debt.